Improve build instructions and reorganize README

bwbohl opened the pull request.

bwbohl committed:
move oXygen build-insturctions to separate file

bwbohl committed:
move command line build-instructions to separate file

bwbohl committed:
move section on cloning this repository to separate file

bwbohl committed:
Update CLONING.md

* add heading
* add linebreak at end of file

bwbohl committed:
Update README.md

* move Sentence about MEI source not being a schema to introduction

bwbohl committed:
Update README.md

* Make the description of MEI Source more precise

bwbohl committed:
Update README.md

* improve section on customizing

bwbohl committed:
Update README.md

* add an external reference to the schema
* add an external reference to the guidelines

bwbohl committed:
Update README.md

* move sentence on building customization locally right after section on CI

bwbohl committed:
Update README.md

* duplicate URLs in “Additional Resources”

bwbohl committed:
Update README.md

* improve reference to other repositories

bwbohl committed:
Update README.md

* Add links to the live Guidelines versions deployed on the MEI website

bwbohl committed:
Update BUILD_COMMANDLINE.md

* add heading

bwbohl committed:
Update BUILD_COMMANDLINE.md

* fix numbering of steps

bwbohl committed:
Update BUILD_COMMANDLINE.md

* consistent spelling of Apache Ant

bwbohl committed:
Update README.md

* Update section on Building MEI
** Add basic information on build steps
** Add a list of prerequisites
* Reference environment-specific build instructions

musicEnfanthen commented on README.md:
```diff
@@ -62,175 +66,48 @@ The MEI takes a different approach. With the customization system, schemas may b
 
-   * Build guidelines HTML:
+1. Java (at least Java8 at runtime to support Apache Ant™)
+2. Apache Ant (recommended version: 1.10.13)
```
the build readme speaks of recommended version `1.10.19`

musicEnfanthen commented on README.md:
````diff
@@ -62,175 +66,48 @@ The MEI takes a different approach. With the customization system, schemas may b
-     ```shell
-     ant init
-     ```
+Generally, we try to support the building process on the most common operating systems: Microsoft Windows, Apple macOS and Linux. While it is possible to achieve the above steps with other implementations as those listed as ‘Recommended implementations’, we strongly advise
````
Something missing at the end here?

musicEnfanthen commented on README.md:
````diff
@@ -62,175 +66,48 @@ The MEI takes a different approach. With the customization system, schemas may b
-     ```shell
-     cd [YOUR-CLONE-LOCATION]
-     ```
+Generally, the build process for MEI artifacts is driven by [Apache Ant™](https://ant.apache.org), a Java library and command line tool. Depending on what artifacts you want to build the prerequisites against your operating will increase.
````
Something is missing in: 

`the prerequisites against your operating will increase`

musicEnfanthen replied:
`Generally` is repeated in line 78

musicEnfanthen commented on README.md:
````diff
@@ -62,175 +66,48 @@ The MEI takes a different approach. With the customization system, schemas may b
-     ```
+MEI maintains a Docker image that meets all the prerequisites and can be used for building the artifacts via the command line. How to use the Docker image is described in the [README](https://github.com/music-encoding/docker-mei#readme) of the [docker-mei repository](https://github.com/music-encoding/docker-mei).
+If your build system meets the above prerequisites you can build the artifacts as documented in [Building MEI on the command line](BUILD_COMMANDLINE.md).
+When the command line is not your preferred tool and you do not want to build the Guidelines PDF, you might consider using alternative environments, such as Synchrosoft’s oXygen XML software family. A description of how to set up corresponding transformation scenarios can be found in [Using Oxygen to Build MEI](BUILD_OXYYGEN.md).
````
Maybe rename the file to `Building MEI with Oxygen` for consistency?

musicEnfanthen commented on BUILD_OXYGEN.md:
```diff
@@ -0,0 +1,51 @@
+### Using Oxygen to Build MEI
```
Maybe rename to `Building MEI with Oxygen` for consistency?

musicEnfanthen commented on CLONING.md:
````diff
@@ -0,0 +1,35 @@
+
+    ```shell
+    git clone https://github.com/music-encoding/music-encoding.git --recursive
+    ```
````
Maybe add a short hint that git handles the initialization and update of the submodule and you dont need the following steps (don't you?)

bwbohl replied:
added something, please revisit

musicEnfanthen commented on CLONING.md:
```diff
@@ -0,0 +1,35 @@
+
+## Creating a recursive clone of this repository
+
+Git offers a mechanism called `submodule` that allows you to reference third-party code used in your own project without including the code in your repository. This mechanism is used in the music-encoding repository in order to include the [TEI Stylesheets](https://github.com/TEIC/Stylesheets); these are needed for transforming the ODD files, e.g. to RNG schema files. Cloning a repository includi ...
```
```suggestion
To include the [TEI Stylesheets](https://github.com/TEIC/Stylesheets) which are needed for transforming the ODD files, e.g. to RNG schema files, the music-encoding repository uses a mechanism provided by Git that is called `submodule`. It allows you to reference third-party code used in your own project without including the code in your repository.

Cloning a repository including the referenced submodules is referred to as _creating a recursive clone_.
```

musicEnfanthen commented on BUILD_OXYGEN.md:
```diff
@@ -0,0 +1,51 @@
+
+7. Now, you will be back in your _Configure Transformation Scenario(s)_ window. In the _Global_ section of the window, you will find your new _XML transformation with XSLT_ with the name you gave it in step 4. Click on it and then click on the _Applied associated_ button at the left-bottom corner of your window. This will generate the HTML document for the guidelines.
+
+After a few minutes, the results of this build can be found in the web folder (`music-encoding/source/web`). The guidelines are stored in the `index.html` file.
```
Shouldn't this go into the dist folder to be consistent with the ant build?

bwbohl replied:
thx for spotting just copy-pasted it, without re-reading

bwbohl replied:
Probably we should address this later and when doing so, rework the workflow description to use the `build.xml` and TEI Stylesheets of this repo instead of using the default shipping with oXygen.

bwbohl replied:
cf. #1117

bwbohl committed:
Update CLONING.md

* rephrase paragraph on submodule

Co-authored-by: Stefan Münnich <stefan.muennich@unibas.ch>

bwbohl committed:
Update BUILD_COMMANDLINE.md

fixup of Apache Ant
* trademark
* and recommended version

bwbohl committed:
Update README.md

* improve section on Building MEI

bwbohl committed:
Update BUILD_COMMANDLINE.md

* add section on prerequisites

bwbohl committed:
Update BUILD_OXYGEN.md

* rephrase title

lpugin commented on BUILD_COMMANDLINE.md:
```diff
@@ -1,6 +1,21 @@
+   |----|----------------|
+   |Java| 17 (at least Java 8 at runtime to support Apache Ant™)|
+   |Apache Ant|1.10.13|
+   |Verovio Toolkit|3.9|
```
Why are you recommending 3.9 and not a more recent version ?

bwbohl replied:
Just the latest successfully tested at the moment. Haven't tested later releases at the moment. Feel free to test it with a local build of the mei-docker container an file a pull request. That would be great!

bwbohl replied:
N.B. pull request both here and in docker-mei; testing at least in docker-mei, If you want, also with in a native command line scenario

lpugin replied:
OK. I will see if I can manage to run it in docker-mei, but I will probably wait until the docs are finalized. Thanks for doing that!

bwbohl committed:
Update CLONING.md

add a short hint on the `--recursive` git pull and submodules

bwbohl committed:
Update README.md

improve description and link for xmllint

bwbohl committed:
Update README.md

unify spelling of `command line`

bwbohl committed:
Update README.md

* correct article usage

bwbohl committed:
Update CONTRIBUTING.md

fix punctuation

bwbohl committed:
Update CONTRIBUTING.md

improve wording

bwbohl committed:
Update CONTRIBUTING.md

improve wording

bwbohl committed:
Update CONTRIBUTING.md

make spelling of GitHub consistent

bwbohl committed:
Update CONTRIBUTING.md

fix spelling

bwbohl committed:
Update CONTRIBUTING.md

unify spelling of GitHub

bwbohl committed:
Update CONTRIBUTING.md

fix spelling

bwbohl committed:
Update CONTRIBUTING.md

Remove the sentence on "future plan". Documentation should describe the current status, future plans should go into issues or discussions.

bwbohl committed:
Update CONTRIBUTING.md

* remove preposition

bwbohl committed:
Update CONTRIBUTING.md

unify spelling of backward-incompatible

bwbohl committed:
Update CONTRIBUTING.md

* fix sentence

bwbohl committed:
add posix conformant newline at file-end

bwbohl committed:
Update README.md

fix links

bwbohl committed:
Update README.md

add whitespace

bwbohl committed:
Update README.md

* remove paragraph on build system prerequisites

bwbohl committed:
Update build.xml

* update references xerces version
closes #1111

bwbohl committed:
build.xml: remove deprecated comment

remove deprecated comment

bwbohl committed:
build.xml: remove whitespaces in empty lines

bwbohl commented:
this should be ready to merge for now. Some issues should be fixed afterwards.

bwbohl committed:
Update BUILD_OXYGEN.md

* rephrase

bwbohl committed:
Update BUILD_OXYGEN.md

* fix spelling

musicEnfanthen commented:
Should be OK to go into the guidelines-test branch although it will need some more fine-tuning (cf. #1117).

musicEnfanthen commented on CONTRIBUTING.md:
```diff
@@ -30,7 +30,7 @@ Pull requests containing contributions of any significant changes should be acco
 ### Customizations and Tests
 
-Customizations to the schema should be placed in the `customizations` folder, named as `mei-NAME-[suffix].xml`. Our automated build system, TravisCI, will build all customizations in this folder and ensure there are no problems. It will also run our automated tests to ensure no validation problems.
+Customizations to the schema should be placed in the `customizations` folder, and named according to the scheme: `mei-NAME-`[suffix].xml`. Our automated build system, TravisCI, will build all customizations in this folder and ensure there are no problems. It will also run our automated tests to ensure no validation problems.
```
Reference to TravisCI should be removed or adjusted?

musicEnfanthen replied:
Fixed by @ahankinson in 2bebf3c63183819a20bcbe232b91ffeca9528999
